api/include/opentelemetry/logs/logger.h

```diff
@@ -43,81 +60,105 @@ class Logger
-   * A default LogRecord that will be assigned if no parameters are passed to Logger's .log() method
-   * which should at minimum assign the trace_id, span_id, and timestamp
+  /**
+   * The base Log(...) method that all other Log(...) overloaded methods will eventually call,
+   * in order to create a log record.
+   *
+   * Note this takes in a KeyValueIterable for the resource and attributes fields.
    */
-  virtual void log(const LogRecord &record) noexcept = 0;
-
-  /** Overloaded methods for unstructured logging **/
-  inline void log(nostd::string_view message) noexcept
+  virtual void Log(core::SystemTimestamp timestamp,
+                   Severity severity,
+                   nostd::string_view name,
+                   nostd::string_view body,
+                   const common::KeyValueIterable &resource,
+                   const common::KeyValueIterable &attributes,
+                   trace::TraceId trace_id,
+                   trace::SpanId span_id,
+                   trace::TraceFlags trace_flags) noexcept = 0;
+
+  /*** Overloaded methods for KeyValueIterables ***/
+  /**
+   * The secondary base Log(...) method that all other Log(...) overloaded methods except the one
+   * above will eventually call,  in order to create a log record.
+   *
+   * Note this takes in template types for the resource and attributes fields.
+   */
+  template <class T,
+            class U,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<U>::value> * = nullptr>
+  void Log(core::SystemTimestamp timestamp,
+           Severity severity,
+           nostd::string_view name,
+           nostd::string_view body,
+           const T &resources,
+           const U &attributes,
+           trace::TraceId trace_id,
+           trace::SpanId span_id,
+           trace::TraceFlags trace_flags) noexcept
   {
-    // Set severity to the default then call log(Severity, String message) method
-    log(Severity::kDefault, message);
+    Log(timestamp, severity, name, body, common::KeyValueIterableView<T>(resources),
+        common::KeyValueIterableView<U>(attributes), trace_id, span_id, trace_flags);
   }
 
-  inline void log(Severity severity, nostd::string_view message) noexcept
+  void Log(core::SystemTimestamp timestamp,
+           Severity severity,
+           nostd::string_view name,
+           nostd::string_view body,
+           std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>> resource,
+           std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>> attributes,
+           trace::TraceId trace_id,
+           trace::SpanId span_id,
+           trace::TraceFlags trace_flags) noexcept
   {
-    // TODO: set default timestamp later (not in API)
-    log(severity, message, core::SystemTimestamp(std::chrono::system_clock::now()));
+    return this->Log(timestamp, severity, name, body,
```

Comment:
Is there a practical reason why you need `this->` here ?

Reply:
Not really, but I thought it would be better to follow the convention of [StartSpan()](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/trace/tracer.h#L52)